### ISSUE TITLE
[Improvement] Add check mechanism to disable `assert` in Gravitino

### ISSUE THREAD

**Issue by yuqi1129** [datastrato/gravitino#2382]
### What would you like to be improved?

We should explicctly disable using `assert` in Gravition and throw an exception if it violate the rules

**Comment by FrankYang0529** [comment]
Hi @yuqi1129, is this issue about adding [AssertFalse](https://errorprone.info/bugpattern/AssertFalse) errorprone? If yes, I would like to take it. Thank you.

**Comment by coolderli** [comment]
> Hi @yuqi1129, is this issue about adding [AssertFalse](https://errorprone.info/bugpattern/AssertFalse) errorprone? If yes, I would like to take it. Thank you.

@FrankYang0529  I think it is related to https://github.com/datastrato/gravitino/pull/2375.

**Comment by yuqi1129** [comment]
> Hi @yuqi1129, is this issue about adding [AssertFalse](https://errorprone.info/bugpattern/AssertFalse) errorprone? If yes, I would like to take it. Thank you.

I don't think so,  what we are going to do is to forbid the syntax `assert condition`.
